core/src/main/java/tech/tablesaw/api/Table.java

```diff
@@ -203,6 +203,49 @@ public DataFrameWriter write() {
         return new DataFrameWriter(defaultWriterRegistry, this);
     }
 
+    private Table padColumnsWithMissing(boolean addColumns, final Column<?>... cols) {
+        int max = 0;
+        for (Column<?> col : columnList) {
+            max = Math.max(max, col.size());
+        }
+        for (Column<?> col : cols) {
+            max = Math.max(max, col.size());
+        }
+        for (Column<?> col : columnList) {
+            while (col.size() < max) {
+                col.appendMissing();
+            }
+        }
+        for (Column<?> col : cols) {
+            while (col.size() < max) {
+                col.appendMissing();
+            }
+        }
+        if (addColumns) {
+            addColumns(cols);
+        }
+        return this;
+    }
+
+    /**
+     * Appends missing values to columns in this table,
+     * to ensure the table is rectangular.
+     * @return this Table
+     */
+    public Table padColumnsWithMissing() {
+        return padColumnsWithMissing(false);
+    }
+
+    /**
+     * Appends missing values to columns in this table and to the columns given as arguments,
+     * before adding the columns. This ensures the table is rectangular.
+     * @param cols the columns to pad and add 
+     * @return this Table
+     */
+    public Table padWithMissingAndAddColumns(final Column<?>... cols) {
+        return padColumnsWithMissing(true, cols);
```

Comment:
I think you could simplify `padColumnsWithMissing` by making it just act on the table and take no args if you add the columns first before calling like:

```
    addColumns(cols);
    return padColumnsWithMissing();
```

Reply:
Actually, I think I'd remove this method. It's not much harder to call `addColumns(cols).padColumnsWithMissing();` vs `padWithMissingAndAddColumns(cols)`

Reply:
But isn't the point that you want to pad _before_ adding, since addColumns throw an Exception if the column are not properly sized? The non-args padColumnsWithMissing() is for the case where the column sizes have been changed after they've been added.

Reply:
Ah, good point. I might rename to `addColumnsAndRectangularize` so that if you're auto-completing or looking through the javadocs it comes up right next to `addColumns`. That name still seems long to me though...  Maybe `addColumns(boolean rectangularize, Column<?>... cols)`?

Reply:
@hallvard I don't think that's really the point. 

The exception is good because adding a column of a different size is almost always a logic error. Most new columns are derived from existing ones, so you rarely intentionally add a column that aligns perfectly, except that it doesn't have a value for some contiguous block of rows at the end of the table.  In other words, appending missing is rarely the fix you want.

On the other hand, forgetting to add values to a column inside of a table _is_ a frequent programming mistake where appending a missing value is often the right solution. So that's the (second) problem we're trying to solve.

The first problem was to prevent toString() from blowing up. I think we're trying to do too many things here. Maybe we should just merge the toString fix and leave all the other stuff separate until we agree.

Reply:
@benmccann 
> Actually, I think I'd remove this method. It's not much harder to call addColumns(cols).padColumnsWithMissing() vs padWithMissingAndAddColumns(cols) (or addColumns(cols).rectangularize() if we rename the method)

I don't really like rectangularize, because I think we're trying to prevent tables from becoming non-rectangular to start with. 

This is what I've been asking for from the beginning. 

On the very rare occasion that you have n different columns of n different lengths and want them all in a single table of one length, you can write a single while loop.